configure.ac

```diff
@@ -1878,6 +1879,24 @@ AS_CASE([$enable_ocamltest,OCAML__DEVELOPMENT_VERSION],
   [yes,*|,true],[ocamltest='ocamltest'],
   [ocamltest=''])
 
+AS_IF([test x"$ocamltest" = "xocamltest"],
+  [AC_CHECK_PROGS([DIFF], [patdiff diff], [none])
+  DIFF_FLAGS=""
+  flags=""
+  AS_CASE([$DIFF],
+    [diff], [flags="--strip-trailing-cr -u"],
+    [none],
+      [AC_MSG_ERROR([ocamltest requires a diff tool])])
+  for flag in ${flags}; do
+    AC_MSG_CHECKING([whether $DIFF supports $flag]);
+    if [$DIFF $DIFF_FLAGS $flag $0 $0 > /dev/null 2>&1]; then
```

Comment:
Please could this be done with `AS_IF` - I found the shell `if` with a m4sh `[` initially confusing.